chirho/dynamical/handlers/trace.py

```diff
@@ -9,7 +10,43 @@
 T = TypeVar("T")
 
 
+@functools.singledispatch
+def append(fst, rest: T) -> T:
```

Comment:
Should this be `append(fst: T, rest: T) -> T:`?

Reply:
No, that doesn't work with `singledispatch`, which expects registered types to be subtypes of the original type on the first argument.  It's an annoying aspect of the way `singledispatch` interacts with `mypy`.